dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/JdbcAnalyticsManager.java

```diff
@@ -411,6 +393,62 @@ private String getFromWhereClause( DataQueryParams params, String partition )
     }
 
     /**
+     * Generates a subquery if preAggregationMeasureCriteria is given
+     * returns the partition of not.
+     */
+    private String getPartition( DataQueryParams params, String partition )
+    {
+
+        if(params.isDataType( DataType.NUMERIC ) && !params.getPreAggregateMeasureCriteria().isEmpty())
+        {
+            SqlHelper sqlHelper = new SqlHelper();
+            String sql = "";
+
+            sql += "(select * from " + partition + " ";
+
+            for( MeasureFilter filter : params.getPreAggregateMeasureCriteria().keySet() )
+            {
+                Double criterion = params.getPreAggregateMeasureCriteria().get( filter );
+
+                if ( EQ.equals( filter ) )
```

Comment:
Instead of IF-ELSE proliferation, build and use a static map of Java operator > SQL operator. Reduce to a one-liner.